tests/components/met/test_weather.py

```diff
@@ -36,6 +48,29 @@ async def test_legacy_config_entry(
     assert len(er.async_entries_for_config_entry(entity_registry, entry.entry_id)) == 1
 
 
+async def test_weather(hass: HomeAssistant, mock_weather) -> None:
+    """Test states of the weather."""
+
+    await init_integration(hass)
+    assert len(hass.states.async_entity_ids("weather")) == 1
+    entity_id = hass.states.async_entity_ids("weather")[0]
+
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == ATTR_CONDITION_CLOUDY
+    assert state.attributes[ATTR_WEATHER_TEMPERATURE] == 15
+    assert state.attributes[ATTR_WEATHER_PRESSURE] == 100
+    assert state.attributes[ATTR_WEATHER_HUMIDITY] == 50
+    assert state.attributes[ATTR_WEATHER_WIND_SPEED] == 10
+    assert state.attributes[ATTR_WEATHER_WIND_BEARING] == 90
+    assert state.attributes[ATTR_WEATHER_DEW_POINT] == 12.1
+    assert state.attributes[ATTR_WEATHER_UV_INDEX] == 1.1
```

Comment:
I looked into it, but felt like it was on the edge where it was better to go the individual attributes route, since the `forecast` isn't being tested. Maybe improving the test coverage should be part of a separate PR?

Reply:
But the UV index isn't part of the forecast object right, so testing that would be outside of the scope anyway

Reply:
Hmmmm... Actually, `uv_index` gets included in the forecast object if it's available, so it's fair to say it's in scope.

There's not tests for `hourly` or `daily` forecasts right now though, and that might be a sizable chunk that'll exercise a bunch of bits I haven't touched here. Would you rather I follow-up with another PR adding tests for the forecasts and updating to the `snapshot_platform` or bundle those changes here as well?

Reply:
Sounds good